index.js

```diff
@@ -1,33 +1,46 @@
 import {spawn} from 'node:child_process';
+import {once} from 'node:events';
 import {lineIterator, combineAsyncIterables} from './utilities.js';
 
 export default function nanoSpawn(command, arguments_ = [], {signal, timeout, nativeOptions} = {}) {
 	const subprocess = spawn(command, arguments_, {...nativeOptions, signal, timeout});
 
-	// eslint-disable-next-line no-async-promise-executor
-	const promise = new Promise(async (resolve, reject) => {
-		try {
-			subprocess.on('close', exitCode => {
-				// TODO: Pass in `stdin` and `stdout` strings here.
-				resolve({
-					exitCode,
-				});
-			});
-
-			subprocess.on('error', error => {
-				reject(error);
-			});
-		} catch (error) {
-			reject(error);
-		}
-	});
+	const promise = getResult(subprocess);
 
 	const stdoutLines = lineIterator(subprocess.stdout);
 	const stderrLines = lineIterator(subprocess.stderr);
-
 	return Object.assign(promise, {
 		[Symbol.asyncIterator]: () => combineAsyncIterables(stdoutLines, stderrLines),
 		stdout: stdoutLines,
 		stderr: stderrLines,
 	});
 }
+
+const getResult = async subprocess => {
+	const result = {};
+	bufferOutput(subprocess, result, 'stdout');
+	bufferOutput(subprocess, result, 'stderr');
+
+	try {
+		const [exitCode, signalName] = await once(subprocess, 'close');
+		return {...getOutput(result), exitCode, signalName};
+	} catch (error) {
+		throw Object.assign(error, getOutput(result));
+	}
+};
+
+const bufferOutput = (subprocess, result, streamName) => {
+	result[streamName] = '';
+	subprocess[streamName].on('data', chunk => {
+		result[streamName] += chunk;
+	});
+};
+
+const getOutput = ({stdout, stderr}) => ({
+	stdout: stripNewline(stdout),
+	stderr: stripNewline(stderr),
+});
+
+const stripNewline = input => input.at(-1) === '\n'
+	? input.slice(0, input.at(-2) === '\r' ? -2 : -1)
+	: input;
```

Comment:
This is _not_ adding an option to disable final newline stripping. 
Not sure whether that's a good idea or not. Please let me know what you think.

Reply:
It can be added if there is an actual need. I prefer to keep it out until then.

Reply:
You mean keeping the option out (as opposed to the newline stripping logic itself), is that correct? If so, sounds good. :+1:

Reply:
Yeah, I meant leaving out the option.